evennia/utils/tests/test_funcparser.py

```diff
@@ -216,12 +238,48 @@ def test_parse(self, string, expected):
         # print(f"time: {(t1-t0)*1000} ms")
         self.assertEqual(expected, ret)
 
-    def test_parse_raise(self):
+    @parameterized.expand(
+        (
+            "Test malformed This is $dummy(a, b) and $bar(",
+            "Test $funcNotFound()",
+        )
+    )
+    def test_parse_raise_unparseable(self, unparseable):
         """
         Make sure error is raised if told to do so.
 
         """
-        string = "Test malformed This is $dummy(a, b) and $bar("
+        with self.assertRaises(funcparser.ParsingError):
+            self.parser.parse(unparseable, raise_errors=True)
+
+    @patch("evennia.utils.funcparser._MAX_NESTING", 2)
+    def test_parse_max_nesting(self):
+        """
+        Make sure it is an error if the max nesting value is reached.
+
+        TODO: Does this make sense? When it sees the first function, len(callstack)
+        is 0. It doesn't raise until the stack length is greater than the
+        _MAX_NESTING value, which means you can nest 4 values with a value of
+        2, as demonstrated by this test.
+        """
+        string = "$add(1, $add(1, $add(1, $add(1, $toint(42)))))"
+        ret = self.parser.parse(string)
+
+        # TODO: Does this return value actually make sense?
+        # It removed the spaces from the calls.
+        self.assertEqual("$add(1,$add(1,$add(1,$add(1,$toint(42)))))", ret)
+
+        with self.assertRaises(funcparser.ParsingError):
+            self.parser.parse(string, raise_errors=True)
+
+    def test_parse_underlying_exception(self):
+        string = "test $add(1, 1) $raise()"
+        ret = self.parser.parse(string)
+
+        # TODO: Does this return value actually make sense?
+        # It completed the first function call.
+        self.assertEqual("test 2 $raise()", ret)
```

Comment:
My guess would be that `$raise` is not a recognized funcparser func, so it's not parsed. If so, that's what you are catching in your `assertRaises` below.

Reply:
Actually, I added `_raises_callable` above on line 81, and registered it with the test parser on line 97, so this is WAI.

Reply:
Updated the `_raises_callable` to throw `RuntimeException`, to make it more clear that we're testing the exception thrown by the underlying function. Also will make the test more resilient.